scripts/gen_binding.py

```diff
@@ -444,16 +481,16 @@ def normalize_arg(arg):
     js_impl_full = "\n".join(js_impl)
     wrap_args = list(zip(js_args, js_arg_types))
     if methods_only:
-        wrap_args = [('this', 'tensor')] + wrap_args
-    js_ptr_args = [x[0] if x[1] != "tensor" else f"{x[0]}.ptr" for x in wrap_args]
+        wrap_args = [('this', 'Tensor')] + wrap_args
+        js_tensor_args = ["this"] + js_tensor_args
+    js_ptr_args = [x[0] if x[1] != "Tensor" else f"{x[0]}.ptr" for x in wrap_args]
     js_ptr_result = f"const _ptr = fl._{op}({', '.join(js_ptr_args)})"
-    js_tensor_args = [f"{arg[0]}" for arg in wrap_args if arg[1] == "tensor"]
-    js_provenance_args = '||'.join([f"{t}.provenance" for t in js_tensor_args])
-    js_requires_grad_args = '||'.join([f"{t}.requires_grad" for t in js_tensor_args])
+    js_provenance_args = '||'.join([f"{t}.provenance" for t in js_tensor_args] + [f"{tv}.some((t) => t.provenance)" for tv in js_tensor_vector_args])
```

Comment:
provenance actually contains information about where the tensor is from (non-zero hash of root device and populated during network transfer).  The original logic preserves the information by short circuiting, whereas `some` generates a boolean.

The `requires_*` are logically sound, but I believe early benchmarking showed the JSC JIT was able to handle explicit chained `||` faster than lambda evaluation.  I know it's ugly, but it's the only way I was able to get performance (this code is pretty hot).

Please revert these bits :)

Reply:
I don't think we can generate explicit `||`s for the members of a TensorVector at codegen time since we don't know the length of the vector.

Is a for loop more performant than lambdas? I should be able to use a loop to preserve the return type of provenance as well.

Reply:
hmm good point on generating explicit `||` for TensorVector - we can use `tensor_vector.reduce((x,y) => x || y)` and leave the other args as is